### ISSUE TITLE
Embedding Dashboard

### ISSUE THREAD

**Issue by krishanthvs** [apache/superset#30544]
### Bug description

I have created the dashboard. The share option gives the perm link. When i am sharing this to the user who does not have superset access its asking for authentication. How to bypass this. And give the dashboard link without the edit button

### Superset version

master / latest-dev

### Python version

3.11

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by krishanthvs** [comment]
@dosu can you please link this to someone who has expertise on the same

**Comment by krishanthvs** [comment]
@rusackas can you please help me on this

**Comment by apoteet** [comment]
The bot is correct about the steps you need to do. [This article](https://www.tetranyde.com/blog/embedding-superset) provides some more detail on _how_ to implement fetchGuestToken. Start from `Approach 1 - Using Superset Embedded SDK`

**Comment by krishanthvs** [comment]
@apoteet  cannot this be done with applying the required permissions to the public role

**Comment by apoteet** [comment]
Correct. For embedded dashboards to work, you need to fetch a guest token and provide it to the dashboard. This guest token has the `public` role, so it needs all the permissions needed to view dashboards. 
If you're worried about the public internet getting access to your superset, you should deploy superset and your app behind a firewall

**Comment by rusackas** [comment]
I think the last answer pretty much solved this, which is why it's been dormant for >6 months. I'll close it, but I'm happy to revisit/reopen if needed.
